shell/platform/common/client_wrapper/core_implementations.cc

```diff
@@ -195,9 +195,32 @@ bool TextureRegistrarImpl::MarkTextureFrameAvailable(int64_t texture_id) {
       texture_registrar_ref_, texture_id);
 }
 
+void TextureRegistrarImpl::UnregisterTexture(int64_t texture_id,
+                                             std::function<void()> callback) {
+  if (callback == nullptr) {
+    FlutterDesktopTextureRegistrarUnregisterExternalTexture(
+        texture_registrar_ref_, texture_id, nullptr, nullptr);
+    return;
+  }
+
+  struct Captures {
+    std::function<void()> callback;
+  };
+  auto captures = std::make_unique<Captures>();
+  captures->callback = callback;
+  FlutterDesktopTextureRegistrarUnregisterExternalTexture(
+      texture_registrar_ref_, texture_id,
+      [](void* opaque) {
+        auto captures = reinterpret_cast<Captures*>(opaque);
+        captures->callback();
+        delete captures;
+      },
+      captures.release());
```

Comment:
The `std::make_unique` and immediate `.release()` seems unnecessary. Consider just `new`ing up `Captures` directly, like this example above:

https://github.com/flutter/engine/blob/6e1c88fa7676566cb051c79ef082d5f4f9ce05ae/shell/platform/common/client_wrapper/core_implementations.cc#L78-L92

Reply:
I'd generally prefer we use `std::unique_ptr` where possible, but since ownership of the captures is getting handed off anyway, agreed it's not adding much. We _should_ be consistent throughout this file though, so agreed I'd just go with new/delete.